### ISSUE TITLE
$params['all'] strips off the leading colon if the first parameter is a trailing parameter

### ISSUE THREAD

**Issue by Renegade334** [phergie/phergie-irc-parser#31]
`$params['all']` should contain a verbatim copy of the parameters string; however, if the first parameter is a trailing parameter with a leading colon, the colon is chomped by `Parser::strip()`. This makes it indiscernible from `$params['all']` whether the parameters string is a set of individual middle parameters, or one trailing parameter.

The following patch would "correct" this behaviour. It affects 42 of the test parser cases. I can't see that anything that currently relies on `$params['all']` would be affected by this change - any use of `$params['all']` would already have to be colon-aware, since colons elsewhere in the params string were still retained, it was just leading colons that were being chomped.

``` diff
diff --git a/src/Parser.php b/src/Parser.php
index 972f190..b299f06 100644
--- a/src/Parser.php
+++ b/src/Parser.php
@@ -407,7 +407,7 @@ class Parser implements ParserInterface
             }
 
             // Clean up and store the processed parameters
-            $params = array_merge(array('all' => $params[0]), array_filter($params));
+            $params = array_merge(array('all' => ltrim($parsed['params'])), array_filter($params));
             $params = $this->removeIntegerKeys($params);
             $parsed['params'] = $params;
         } elseif (ctype_digit($command)) {
@@ -421,7 +421,7 @@ class Parser implements ParserInterface
                 $temp = explode(' ', ltrim($parsed['params']), 2);
                 $parsed['target'] = array_shift($temp);
                 if ($parsed['params'] = (!empty($temp)) ? (' ' . array_shift($temp)) : '') {
-                    $all = $this->strip($parsed['params']);
+                    $all = ltrim($parsed['params']);
                     if (strpos($parsed['params'], ' :') !== false) {
                         list($head, $tail) = explode(' :', $parsed['params'], 2);
                     } else {
```

**Comment by elazar** [comment]
The change you're proposing seems to make sense. If you have any lingering questions or issues, feel free to present them here, or otherwise to file a PR.

**Comment by elazar** [comment]
Merged #33.
